core/model/modx/sources/modfilemediasource.class.php

```diff
@@ -622,13 +641,7 @@ public function renameObject($oldPath,$newName) {
         $newPath = $this->fileHandler->sanitizePath($newName);
         $newPath = dirname($oldPath).'/'.$newPath;
 
-        /* check to see if the new resource already exists */
-        if (file_exists($newPath)) {
-            if (is_dir($newPath)) {
-                $this->addError('name',$this->xpdo->lexicon('file_folder_err_ae'));
-                return false;
-            }
-            $this->addError('name',sprintf($this->xpdo->lexicon('file_err_ae'),$newName));
+        if (!$this->checkObjectExist($newPath,$newName)) {
```

Comment:
After the you adjust the `checkObjectExist` to return `true` if the object exists, this will need to change as well - removing the `!`.